components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.webextensions
+
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.webextension.WebExtension
+import mozilla.components.concept.engine.webextension.WebExtensionDelegate
+
+/**
+ * Provides functionality to make sure web extension related events in the
+ * [Engine] are reflected in the browser and session states by dispatching
+ * the corresponding actions to the [BrowserStore].
+ */
+object WebExtensionSupport {
```

Comment:
I almost feel like this should go into a `feature-webextensions` since this is going to grow and actually provide a `feature`. :)

Reply:
@pocmo yes I was also thinking about that and going back and forth on `feature-webextensions`. I think I'd like to land this to unblock other work and get back to it in the coming weeks.

Reply:
Main reason I decided against introducing a separate `feature` is that if we ultimately (after the migration) end up adding the store to the engine, then we don't need this anymore.

Reply:
I was thinking that the code that will add browser actions will have to live in that feature eventually too. :)
But, yeah, definitely doesn't need to block landing this.